tests/phpunit/includes/abstract-testcase.php

```diff
@@ -142,15 +142,18 @@ public function set_up() {
 	 * After a test method runs, resets any state in WordPress the test method might have changed.
 	 */
 	public function tear_down() {
-		global $wpdb, $wp_query, $wp;
+		global $wpdb, $wp_the_query, $wp_query, $wp;
 		$wpdb->query( 'ROLLBACK' );
 		if ( is_multisite() ) {
 			while ( ms_is_switched() ) {
 				restore_current_blog();
 			}
 		}
-		$wp_query = new WP_Query();
-		$wp       = new WP();
+
+		// Reset query, main query, and WP globals similar to wp-settings.php.
+		$wp_the_query = new WP_Query();
+		$wp_query     = $wp_the_query;
```

Comment:
Is there a good reason to set the `$wp_query` global to the main query? This will result in any use of the global `$wp_query` in tests being treated as the main query. I would find unexpected, since I don't think that was case previously. I think it would be better to set both to separate `WP_Query` objects, like:

```suggestion
		$wp_query     = new WP_Query();
```

Reply:
@joemcgill This is actually the case, as WP's bootstrap process sets the two globals like that. In any regular WordPress request, `$wp_query` can be assumed to be the same as `$wp_the_query`, except when "manually" overwritten e.g. by a plugin. The fact that this wasn't behaving the same in tests before is rather the part that it's broken.

I see your concern here, but given that all tests pass without modifications, it seems no tests was making any specific assumptions about this. And the change here brings parity with how it is expected to behave in a regular WordPress request.

Reply:
Sounds good to me. I do think setting the default state as close to how it's set by the bootstrap process should be the goal.